php/WP_Mock/Hook.php

```diff
@@ -41,16 +41,27 @@ protected function safe_offset($value): string
     {
         if (null === $value) {
             return 'null';
-        /** the following is to prevent a possible return mismatch when {@see \WP_Mock\Functions::type()} is used with 'callable' */
-        } elseif ($value instanceof Closure || Closure::class === $value || (is_string($value) && '<CLOSURE>' === strtoupper($value))) {
+        }
+
+        /** the following is to prevent a possible return mismatch when {@see \WP_Mock\Functions::type()} is used with 'callable' and to correctly create safe offsets for processors when expecting, via {@see WP_Mock\Functions::type(\Closure::class)}, that a hook that uses a closure is added*/
+        $closure = fn() => null;
+        if ($value instanceof Closure || Closure::class === $value || (is_string($value) && '<CLOSURE>' === strtoupper($value)) || ($value instanceof \Mockery\Matcher\Type && $value->match($closure))){
```

Comment:
```suggestion
        /**
         * The following is to prevent a possible return mismatch when {@see Functions::type()} is used with `callable`,
         * and to correctly create safe offsets for processors when expecting that a hook that uses a closure is added via {@see Functions::type(Closure::class)}.
         */
        $closure = fn() => null;
        if ($value instanceof Closure || Closure::class === $value || (is_string($value) && '<CLOSURE>' === strtoupper($value)) || ($value instanceof Type && $value->match($closure))){
```

I suggest to break the comment into multi-lines for clarity - also we can use the opportunity to avoid fully qualified namespaces here